Objective-C/CBLCollection.mm

```diff
@@ -408,6 +406,23 @@ - (BOOL) isValid {
     return [self collectionIsValid: nil];
 }
 
+- (BOOL) isEqual: (id)object {
+    if (self == object)
+        return YES;
+    
+    CBLCollection* other = $castIf(CBLCollection, object);
+    if (!other)
+        return NO;
+    
+    if (!(other && [self.name isEqual: other.name] &&
+          [self.scope.name isEqual: other.scope.name] &&
+          [self.db.path isEqual: other.db.path])) {
+        return NO;
+    }
+    
```

Comment:
I think instead of comparing collection name, scope, and db path, we can compare if they have the same c4col object or not. By doing that, it can fix the issue for the case that the collection is deleted and recreated.